src/jit/compiler.cpp

```diff
@@ -546,13 +546,9 @@ var_types    Compiler::argOrReturnTypeForStruct(unsigned size, CORINFO_CLASS_HAN
             if (size <= MAX_RET_MULTIREG_BYTES)
             {
 #ifdef _TARGET_ARM64_
-                assert(size > TARGET_POINTER_SIZE);
-
-                // For structs that are 9 to 16 bytes in size set useType to TYP_STRUCT, 
-                // as this means a 9-16 byte struct value in two registers
-                //
-                useType = TYP_STRUCT;
-#endif // _TARGET_ARM64_
+                // TODO-ARM64-HFA - Implement x0,x1 returns   
+                // TODO-ARM64     - Implement HFA returns   
+#endif // _TARGET_XXX_
```

Comment:
FEATURE_MULTIREG_RET is zero, so this code doesn't execute
And yes it does the reasonable thing which is not to change useType to TYP_STRUCT